### ISSUE TITLE
Door_Bell.js

### ISSUE THREAD

**Issue by thechrisreese** [sibartlett/homebridge-wink3#36]
[door_bell.js.zip](https://github.com/sibartlett/homebridge-wink3/files/1236100/door_bell.js.zip)

Just trying to see if this works. I cribbed the top part of camera.js for the motion sensor and the battery at the bottom to make a door_bell.js so that the Ring Doorbell motion sensor works.

**Comment by sibartlett** [comment]
I've created a pull request: #37

**Comment by sibartlett** [comment]
@thechrisreese can you try this new version: `1.6.0beta1`?

It includes support for door bells - not just as a motion sensor, but as a door bell too!

**Comment by thechrisreese** [comment]
What what? Ok how do I install it because I haven’t done a beta.

**Comment by sibartlett** [comment]
npm install -g homebridge-wink3@1.6.0beta1

**Comment by thechrisreese** [comment]
Ok tomorrow I’ll test it.

**Comment by thechrisreese** [comment]
@sibartlett I’m not seeing a change from it just being a motion sensor. Will keep trying it.

**Comment by sibartlett** [comment]
Hmm, I added the Doorbell service - I figured it'd allow for automations and notifications when someone presses the button.

**Comment by thechrisreese** [comment]
I will try again. I had to remove Ring from Wink because it kept failing. I will look into it again. 

What did you add to Door_Bell beyond Motion Control? Did you add the button press?

**Comment by sibartlett** [comment]
Theses are the changes I made: https://github.com/sibartlett/homebridge-wink3/blob/79033345984631f0c45268c9b66f9e07e27c6019/src/devices/door_bell.js

**Comment by thechrisreese** [comment]
Yeah ok. That should add the Motion and Button Press. Ok I will try out again tonight. I feel like that should work. Of course I am running things on Docker which may make things odder.

**Comment by sibartlett** [comment]
I use docker too :)

**Comment by thechrisreese** [comment]
Ok!  Well my Docker decided to mess everything up when trying to apt-get install libpcap-dev so I am still unwrapping that mess on my NAS.

**Comment by sibartlett** [comment]
I dunno if it helps, but I found this docker image works well for homebridge: https://hub.docker.com/r/oznu/homebridge/

**Comment by thechrisreese** [comment]
@sibartlett that one is nice because I can figure out where the Plugins install to. However its running an old version of Node which wont let me install a bunch of my favorite plugins.

**Comment by davidvanvo** [comment]
I just tried this and nothing happens when the door bell is pressed. Can confirm that the Wink app shows this event. Motion works.

**Comment by thechrisreese** [comment]
@davidvanvo my doorbell stopped working entirely on wink3. Is your motion still working? I want to fix this now that I have a week off.

**Comment by thechrisreese** [comment]
[2017-12-23 10:37:48] [Wink] Not supported by HomeKit: Home Sprinkler (sprinkler/6577)                                                                                                                                           
[2017-12-23 10:37:48] [Wink] Not supported by HomeKit: Unknown Device (unknown_device/24988)                                                                                                                                     
[2017-12-23 10:37:48] [Wink] Not supported by HomeKit: Front Door (door_bell/39923)  


Ok it is flagging it as not supported by Homekit again....so I have to tweak the Door_bells.js and other files to make it see it. Likely they changed how door_bells are registered via hapnode.js

**Comment by sibartlett** [comment]
Not sure if it’s an issue with hap, more likely an issue with wink3.

Have you got the JSON for the doorbell, that comes from wink?

**Comment by thechrisreese** [comment]
@sibartlett I will have to pull it again. 

I am trying to reconstruct my logic around this from August - 

Where does it pull this information?

 .forEach(device => {
        const definition = this.definitions[device.object_type];

        if (!definition) {
          ignoreDevices.push([device, "Not supported by HomeKit"]);
          return;
        }

Particularly where are the definitions coming from? Is it from doorbells.js?

**Comment by thechrisreese** [comment]
From HapNode.js - I feel this is possibly what is causing the issues

Accessory.Categories = {
  OTHER: 1,
  BRIDGE: 2,
  FAN: 3,
  GARAGE_DOOR_OPENER: 4,
  LIGHTBULB: 5,
  DOOR_LOCK: 6,
  OUTLET: 7,
  SWITCH: 8,
  THERMOSTAT: 9,
  SENSOR: 10,
  ALARM_SYSTEM: 11,
  SECURITY_SYSTEM: 11, //Added to conform to HAP naming
  DOOR: 12,
  WINDOW: 13,
  WINDOW_COVERING: 14,
  PROGRAMMABLE_SWITCH: 15,
  RANGE_EXTENDER: 16,
  CAMERA: 17,
  IP_CAMERA: 17, //Added to conform to HAP naming
  VIDEO_DOORBELL: 18,
  AIR_PURIFIER: 19,
  AIR_HEATER: 20, //Not in HAP Spec
  AIR_CONDITIONER: 21, //Not in HAP Spec
  AIR_HUMIDIFIER: 22, //Not in HAP Spec
  AIR_DEHUMIDIFIER: 23 // Not in HAP Spec
}

**Comment by davidvanvo** [comment]
@thechrisreese mine is still working fine. i’m not running the latest wink3 homekit so not sure if there has been any updates to door_bell.js.
